aten/src/ATen/native/UpSample.h

```diff
@@ -136,10 +142,14 @@ static inline void upsample_3d_shape_check(
       ")");
 
   if (input.defined()) {
+    // Allow for empty batch size but not other dimensions
+    bool valid_empty = false;
+    valid_empty = input.size(0) == 0 && input.size(1) != 0 &&
+      input.size(2) != 0 && input.size(3) != 0 && input.size(4) != 0;
```

Comment:
nit: just assign the expression directly to `valid_empty`, there's no reason to assign it to false first and then reassign